tests/phpunit/tests/kses/wpFilterGlobalStylesPost.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * @group kses
+ *
+ * @covers ::wp_filter_global_styles_post
+ */
+class Tests_Kses_WpFilterGlobalStylesPost extends WP_UnitTestCase {
+
+	/**
+	 * Theme data.
+	 *
+	 * @var array
+	 */
+	private $user_theme_data = array(
+		'isGlobalStylesUserThemeJSON' => 1,
+		'version'                     => 1,
+		'nonSchemaRule'               => 'someValue',
+		'styles'                      => array(
+			'blocks' => array(
+				'core/button' => array(
+					'border' => array(
+						'radius' => '0',
+					),
+				),
+			),
+		),
+	);
+
+	/**
+	 * @ticket 56266
+	 */
+	public function test_should_not_remove_safe_global_style_rules() {
+		$filtered_user_theme_json = $this->filter_global_styles( $this->user_theme_data );
+		$this->assertArrayNotHasKey( 'nonSchemaRule', $filtered_user_theme_json, 'Filtered json data must not contain unsafe global style rules.' );
```

Comment:
I was thinking Line 35 and 43 should be swapped:

- The first function, `should_not_remove`, asserts that `nonSchemaRule` was _removed_, which is the opposite of the name.
- Conversely, the other test, `should_remove`, has an assertion that `styles` was _NOT removed_.

This could just be how I'm reading it.

Reply:
Good catch.
Fixed in f5c52f615ccd2e244206e2cbf65b6e1a59425fef